mods/beds/functions.lua

```diff
@@ -170,13 +170,24 @@ end
 
 -- Public functions
 
+function beds.kick_player(player)
+	lay_down(player, nil, nil, false)
+end
+
 function beds.kick_players()
-	for name, _ in pairs(beds.player) do
-		local player = minetest.get_player_by_name(name)
-		lay_down(player, nil, nil, false)
+	for name in pairs(beds.player) do
+		beds.kick_players(core.get_player_by_name(name))
```

Comment:
Infinite recursion. Did you probably want this?

```suggestion
		beds.kick_player(core.get_player_by_name(name))
```

That aside, I don't think that a separate helper function is needed. If other mods need it, then it should be documented.

Reply:
I think introducing a `kick_player` function, be it exposed or not, is a good idea. Duplicating the `lay_down(player, nil, nil, false)` call (which tbh should probably have been refactored to begin with, but that's another issue) is overall less readable.

I think I'll call it just `kick(player)` to make it a bit more concise and avoid the similarity with `kick_players`.